### PR TITLE
Add back "Final changes to py2help new (#2516)"

### PR DESCRIPTION
**Pull request by couet**

### CODE REVIEW

File: cmake/modules/RootNewMacros.cmake [hunk 1]
```diff
@@ -1449,4 +1449,34 @@ function(find_python_module module)
...
+# library( the executable that includes it or the library that uses it)
+#----------------------------------------------------------------------------
+function(generateHeaders pythonInput output target)
+     add_custom_command(OUTPUT ${output}
+          DEPENDS ${pythonInput} ${CMAKE_SOURCE_DIR}/build/misc/argparse2help.py
+          COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_SOURCE_DIR}/build/misc/argparse2help.py 
+                                       ${pythonInput}
+                                       ${output}
+     )
+     target_sources(${target} PRIVATE ${output})
```
Comment (amadio) [inline cmake/modules/RootNewMacros.cmake]: I believe that the problem is here. CMake is not understanding that `${target}` needs to wait for the custom command to run just from this call to `target_sources()`. We need to use something similar to what is used for the dictionaries. I can reproduce the problem others had on Linux by using ninja as generator. We can add one build with ninja to the PRs if needed for this. (You can also just use `@phsft-bot build just on ubuntu18/gcc7 with flags -G Ninja` or similar).

**Comment by couet** [comment]
@phsft-bot build just on ubuntu18/gcc7 with flags -G Ninja

**Comment by amadio** [comment]
It looks like the bot did not recognize the `-G Ninja` option. It probably only looks into `-D...` options.

**Comment by amadio** [comment]
@phsft-bot build

**Comment by couet** [comment]
@phsft-bot build

**Comment by couet** [comment]
@phsft-bot

**Comment by couet** [comment]
@phsft-bot build

**Comment by couet** [comment]
@phsft-bot build

**Comment by couet** [comment]
@phsft-bot build

**Pull request closed without merging**